yield_plugin/src/yield_plugin.cpp

```diff
@@ -649,6 +696,31 @@ namespace yield_plugin
     return get_collision_time(original_tp, new_list, config_.intervehicle_collision_distance_in_m);
   }
 
+  std::unordered_map<uint32_t, rclcpp::Time> YieldPlugin::get_collision_times_concurrently(const carma_planning_msgs::msg::TrajectoryPlan& original_tp,
+    const std::vector<carma_perception_msgs::msg::ExternalObject>& external_objects)
+  {
+
+    std::unordered_map<uint32_t, std::future<std::optional<rclcpp::Time>>> futures;
+    std::unordered_map<uint32_t, rclcpp::Time> collision_times;
+
+    // Launch asynchronous tasks to check for collision times
+    for (const auto& object : external_objects) {
+      futures[object.id] = (std::async(std::launch::async,[this, &original_tp, &object](){
```

Comment:
You can omit the double parenthesis for lambdas that don't take arguments:
```cpp
futures[object.id] = std::async(std::launch::async, [this, &original_tp, &object] { /* ... */ });
```

Also, you shouldn't need to wrap the `std::async` call in parenthesis.

Reply:
Replaced